test/e2e/framework/internal/junit/junit.go

```diff
@@ -42,5 +47,60 @@ func WriteJUnitReport(report ginkgo.Report, filename string) error {
 		OmitSpecLabels: true,
 	}
 
+	// Sort specs by full name. The default is by start (or completion?) time,
+	// which is less useful in spyglass because those times are not shown
+	// and thus tests seem to be listed with no apparent order.
+	slices.SortFunc(report.SpecReports, func(a, b types.SpecReport) int {
+		res := strings.Compare(a.FullText(), b.FullText())
+		if res == 0 {
+			// Use start time as tie-breaker in the unlikely
+			// case that two specs have the same full name.
+			return a.StartTime.Compare(b.StartTime)
+		}
+		return res
+	})
+
+	// Detect data races in output and mark those tests as failed.
+	// Ginkgo itself captures the warning in the output of those
+	// tests where it is printed, but does not mark the tests as failed.
+	// This is not exactly wrong (the reason might be in code that was
+	// started elsewhere), but without a test being marked as "failed",
+	// prune-junit-xml will prune the output and spyglass won't show
+	// the test as failed.
+	//
+	// Both variants of captured output get checked, just to be on the safe side.
+	for i, spec := range report.SpecReports {
```

Comment:
this seems something that it will run for all jobs once it enabled, can we detect if the golang race detector is enabled and only check this if that is the case?

Reply:
Changed.